cmds.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import "github.com/Azure/azure-docker-extension/pkg/vmextension"
+
+type cmdFunc func(vmextension.HandlerEnvironment) error
+
+var (
+	cmds = map[string]struct {
+		f             cmdFunc // associated function
```

Comment:
Understood.  The compiler can check types, but it cannot do the same for strings.  If you can lean on the compiler to solve this problem I would.